### DISCUSSION TITLE
Add the `$this->addError` API to Livewire Form components (V3)

### DISCUSSION THREAD

**Discussion by WyattCast44** [livewire/livewire#5891] [3. Feature ideas]
Greetings! Working on migrating to V3 and I wanted to propose a feature for the new Form components. 

Currently in V2 (and V3) the Livewire component has the ability to add named errors to the error bag via the `$this->addError()` API. I think this would be awesome to also have on the new Form components. Check out an example usage below, what do you think everyone? I could potentially put together a PR if others like this idea. Cheers!

```php
<?php

namespace App\Livewire\Forms;

use Livewire\Attributes\Rule;
use Livewire\Form;

class Login extends Form
{
    #[Rule(['required', 'email'])]
    public string $email = '';

    #[Rule(['required'])]
    public string $password = '';

    public function attempt(): bool
    {
        $this->validate();

        $status = auth()->attempt(
            credentials: [
                'email' => $this->email,
                'password' => $this->password,
            ],
            remember: $this->remember
        );

        if (!$status) {
            $this->addError('status', 'Please check your credentials and try again.');

            return false;
        }

        return true;
    }
}
```

**Comment by viliusvsx** [comment]
Yeah, I think this would be useful, but I also noticed, that it is possible to use `$this->addError()` in Form Object using `$this->component->addError($name, $message)`.

**Reply by elcapo** [reply]
Oh! Thanks!

I just found that this approach also facilitates setting errors in synthesizers, for instance in the `set` method:

~~~php
$this->context->component->addError($key, $message);
~~~

**Comment by joshhanley** [comment]
Yeah I think that could be useful. PRs are welcome! It could just be a wrapper around the method @viliusvsx mentioned.
